src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -139,72 +132,116 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
+
+                        int k = i + 1;
+                        for (; k < length; k++)
+                        {
+                            char next = normalizedText[k];
+                            if (next == ch)
+                            {
+                                if (next == '\'')
+                                {
+                                    if (k + 1 >= length || normalizedText[k + 1] != '\'')
+                                    {
+                                        // match with single qutoe ('), stop it.
+                                        break;
+                                    }
+                                    else
+                                    {
+                                        // Unescape the double single quotes as one single quote, and continue
+                                        sb.Append('\'');
+                                        k++;
+                                    }
+                                }
+                                else
+                                {
+                                    // match with double quote ("), stop it.
+                                    break;
+                                }
+                            }
+                            else if (next == '\\')
+                            {
+                                if (ch == '\'')
+                                {
+                                    sb.Append('\\'); // in single quote marked string, \ should escaped to double single slash.
+                                    sb.Append('\\');
+                                }
+                                else
+                                {
+                                    sb.Append(next); // in double quote marked string, \ should not escaped.
+                                }
+                            }
+                            else
+                            {
+                                sb.Append(next);
+                            }
+                        }
+
+                        if (k == length)
+                        {
+                            string errorMessaage = normalizedText.Substring(i);
+                            throw new ODataException(ODataErrorStrings.StringItemShouldBeQuoted(errorMessaage));
+                        }
+
+                        sb.Append('"'); // no matter it's single quote or not, just ending it as double quote.
+                        i = k;
+                        break;
+
+                    case ' ':
+                        // ignore all whitespaces between items
+                        break;
+
+                    case ',':
+                        // for multiple comma(s) between items, for example ('abc',,,'xyz'),
+                        // we let it go and leave the next layer to identify the problem, or remove the duplicated silently?

```

Comment:
> or remove the duplicated silently? [](start = 89, length = 35)

Yes; I would leave it to the next layer to handle, not remove silently.

Reply:
Cool. I will change the comments.

---
In reply to: [370714947](https://github.com/OData/odata.net/pull/1642#discussion_r370714947) [](ancestors = 370714947)